index.js

```diff
@@ -10,8 +10,55 @@ const COLORS = 'https://nt-cdn.s3.amazonaws.com/colors.json';
  * @param compHex filter for complementary color hex code
  * @returns Promise
  */
+
 const fetchColors = ({ name, hex, compName, compHex }) => {
-  throw Error('Not implemented');
+  //empty array to hold the output
+  let arr = [];
+  //fetching colors
+  return (
+    fetch(COLORS)
+      //converting to json for manipulation
+      .then((res) => res.json())
+      //then to map out the response
+      .then((res) => {
+        //if to check the color name includes the name param
+        res.map((colorObj) => {
+          if (
+            name &&
+            colorObj.name.toLowerCase().includes(name.toLowerCase())
```

Comment:
Outsource this comparison to a function. If the requirement changed to be case-sensitive you would have to change the comparison in multiple places